spec/features/user_show_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :feature do
+  let!(:user1) do
+    User.create(
+      name: 'test user1',
+      photo: 'https://images3.alphacoders.com/690/690494.jpg',
+      bio: 'test_bio1',
+      posts_counter: 1
+    )
+  end
+
+  let!(:posts) do
+    [
+      Post.create(author: user1, title: 'Post 1', text: 'Post 1 content', comments_counter: 1, likes_counter: 1),
+      Post.create(author: user1, title: 'Post 2', text: 'Post 2 content', comments_counter: 2, likes_counter: 2),
+      Post.create(author: user1, title: 'Post 3', text: 'Post 3 content', comments_counter: 3, likes_counter: 3),
+      Post.create(author: user1, title: 'Post 4', text: 'Post 4 content', comments_counter: 4, likes_counter: 4)
+    ]
+  end
+
+  describe 'show user and post details' do
+    before(:example) do
+      visit user_path(user1)
+    end
+
+    it 'displays user name' do
+      expect(page).to have_content(user1.name)
+    end
+
+    it 'displays user posts counter' do
+      expect(page).to have_content(user1.posts_counter)
+    end
+
+    it 'displays user bio' do
+      expect(page).to have_content(user1.bio)
+    end
+
+    it 'displays user photo' do
+      expect(page).to have_selector("img[src='#{user1.photo}']")
+    end
+
+    it 'should have a see all posts button' do
+      expect(page).to have_link('See all posts')
+    end
+
+    it 'should redirect to the user posts page when the see all posts button is clicked' do
+      click_link('See all posts')
+      expect(page).to have_current_path(user_posts_path(user1))
+    end
+
+    it 'displays user\'s first 3 posts' do
+      posts.last(3).each do |post|
+        expect(page).to have_content(post.title)
+        expect(page).to have_content(post.text)
+      end
+    end
+
+    it 'displays user posts' do
+      expect(page).to have_content('Post 3')
+      expect(page).to have_content('Post 4')
```

Comment:
Nice job so far👍🏼. As a [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md), let's add a test that redirects to the post's show page when you click on a user's post😉. It seems you forgot to implement this in your `user_show_spec.rb` please do well to implement that😉.


<img width="523" alt="image" src="https://github.com/ReemMohamedAbdelfatah/rails-blog-app/assets/64643995/478d2fae-a073-4b09-ba6c-8b76dd91bd2e">